plugin.json

```diff
@@ -0,0 +1,36 @@
+{
+  "id": "marketplace-addon",
+  "name": "Marketplace Addon",
+  "description": "Automatically update installed plugins to latest versions",
+  "version": "0.1.0",
+  "min_server_version": "5.14.0",
+  "server": {
+    "executables": {
+        "linux-amd64": "server/dist/plugin-linux-amd64",
+        "darwin-amd64": "server/dist/plugin-darwin-amd64",
+        "windows-amd64": "server/dist/plugin-windows-amd64.exe"
+    }
+  },
+  "settings_schema": {
+    "settings": [{
+      "key": "NotificationChannelName",
+      "display_name": "Notification Channel",
+      "help_text": "A channel name to send notifications related to plugin updates.",
+      "type": "text"
+    },{
+      "key": "MarketplaceAPIAddress",
+      "display_name": "Marketplace API's Address",
+      "type": "text",
+      "help_text": "Marketplace API used to fetch latest plugins from the Marketplace.",
+      "placeholder": "https://api.integrations.mattermost.com",
+      "default": "https://api.integrations.mattermost.com"
+    },{
+      "key": "UpdateCheckFrequency",
```

Comment:
I made a similar comment on the TOPDF plugin, but I'm wondering if we should just stick to seconds here for consistency in the system console. That being said, I'm actually really liking the flexibility here. Maybe we could investigate adding a new configuration type for duration in the future to streamline all of this.

Reply:
Thanks! I'll update to use seconds here.

`time.Duration` is actually an `int64` and integer values inside JSON are able to unmarshal to `time.Duration` since this is handled by the reflection level at _encoding/json_ pkg.

However, as you pointed out, there is no support for numbers in `manifest.settings_schema.settings.type` for now and thus, string values cannot be recognized and converted to `time.Duration` by default. This is why we had to come up with `xtime.Duration` and a custom unmarshaler for [string durations](https://github.com/ilgooz/mattermost-plugin-marketplace-addon/blob/aedc070/server/x/xtime/duration.go#L19).

Let's definitely support more types in the settings to have a better experience while parsing configurations to structs.